Shared/Realm.Sync.Shared/Session.cs

```diff
@@ -17,27 +17,89 @@
 ////////////////////////////////////////////////////////////////////////////
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
 
 namespace Realms.Sync
 {
     public enum SessionState
     {
-        Authenticating,
-        Binding,
-        Bound,
-        Initial,
-        Stopped,
-        Unbound
+        WaitingForAccessToken = 0,
+        Active,
+        Dying,
+        Inactive,
+        Error,
     }
 
     public class Session
     {
-        public SyncConfiguration Configuration { get; private set; }
+        private static readonly ConcurrentDictionary<SessionHandle, Session> _sessions = new ConcurrentDictionary<SessionHandle, Session>(new SessionHandle.Comparer());
 
-        public Uri ServerUri { get; private set; }
+        private List<ErrorEventArgs> _aggregatedErrors = new List<ErrorEventArgs>();
 
-        public SessionState State { get; private set; }
+        private event ErrorEventHandler _error;
 
-        public User User { get; private set; }
+        public event ErrorEventHandler Error
```

Comment:
PCL doesn't have ErrorEventHandler, so we should opt for a `EventHandler<Exception>`